### PR TITLE
Withdraw the candidate application after feedback

### PR DESCRIPTION
**Pull request by CatalinVoineag**
## Context

We want to make it mandatory to provide feedback when withdrawing an application.

To do this, the application will be withdrawn after the withdrawal feedback has been submitted.

The WithdrawalForm that was previously only saving the feedback will now withdraw the application and submit the feedback.

## Changes proposed in this pull request

Renamed WithdrawalFeedbackForm to WithdrawalForm
Withdraw the application in WithdrawalForm
Add presence validation on selected reason for withdrawal

## Guidance to review

Go to the review app and withdraw a candidate application.

## Link to Trello card

https://trello.com/c/wfjkqik7/832-make-post-withdrawal-reason-non-optional?filter=member:catalinvoineag1

## Screenshots

https://github.com/DFE-Digital/apply-for-teacher-training/assets/11318084/5a63e996-b4bf-4db2-b6da-2900fe0d33b9

### CODE REVIEW

**Comment by CatalinVoineag** [comment]
[Make post-withdrawal reason non-optional](https://trello.com/c/wfjkqik7/832-make-post-withdrawal-reason-non-optional)

File: spec/system/candidate_interface/continuous_applications/withdraws/candidate_withdraws_spec.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 require 'rails_helper'
 
-RSpec.feature 'A candidate withdraws their application', :bullet do
```
Comment (elceebee) [inline spec/system/candidate_interface/continuous_applications/withdraws/candidate_withdraws_spec.rb:3]: I don't think I see an unhappy path scenario (click continue, see errors) -- would be nice to include one (if I've missed it ignore this!)

**Review by elceebee — APPROVED** [review]
Nice work! Tested locally and it works as expected. If there isn't an unhappy path scenario, would be good to add one, but otherwise 🚀

File: spec/system/candidate_interface/continuous_applications/withdraws/candidate_withdraws_spec.rb (hunk 1 above)
Comment (CatalinVoineag) [reply spec/system/candidate_interface/continuous_applications/withdraws/candidate_withdraws_spec.rb:3]: I don't think you can get errors if you click `continue` it's basically a link_to to the feedback form.

You can get errors on the feedback form though, so I'll put in a test for that

**Review by dcyoung-dev — APPROVED** [review]

**Pull request closed without merging**